src/main/java/org/folio/rest/impl/FinanceHelper.java

```diff
@@ -73,10 +80,31 @@
       .map(Pair::getLeft)
       .collect(toList());
     return setCurrentFiscalYear(encumbrances)
-      .thenCompose(aVoid -> createSummary(compPo.getId(), encumbrances.size()))
-      .thenCompose(vVoid -> createEncumbrances(encumbranceDistributionPairs));
+      .thenCompose(v -> checkEnoughMoneyForTransactions(encumbrances))
+      .thenCompose(v -> createSummary(compPo.getId(), encumbrances.size()))
+      .thenCompose(summaryId -> createEncumbrances(encumbranceDistributionPairs));
   }
 
+  private CompletableFuture<Void> checkEnoughMoneyForTransactions(List<Transaction> encumbrances) {
+    Map<String, List<Transaction>> groupedByFund = encumbrances.stream()
+      .collect(groupingBy(Transaction::getFromFundId));
+
+    Map<String, Double> transactionAmountsByFunds = encumbrances.stream()
+      .collect(Collectors.groupingBy(Transaction::getFromFundId, Collectors.summingDouble(Transaction::getAmount)));
```

Comment:
Please use Collector from Java Money API to avoid inaccuracies in the calculations.

Reply:
leveraged java money api